### PR TITLE
Getting an error from aws_sagemaker_notebook_instance table.hydrate call listAwsSageMakerNotebookInstanceTags failed with panic interface conversion. closes #364

### LINKED ISSUE #364: Getting an error from `aws_sagemaker_notebook_instance` table. Please see the detail below.
**Describe the bug**

Create a notebook instance in one of the configured region.
Ran the below query and got that error

```
select * from aws_sagemaker_notebook_instance;
Error: hydrate call listAwsSageMakerNotebookInstanceTags failed with panic interface conversion: interface {} is *sagemaker.NotebookInstanceSummary, not *sagemaker.DescribeNotebookInstanceOutput

```

**Steampipe version (`steampipe -v`)**
: v0.4.1

**Plugin version (`steampipe plugin list`)**
aws: v0.15.0

### PR DESCRIPTION
**Pull request by Priyanka585464**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/aws_sagemaker_notebook_instance []

PRETEST: tests/aws_sagemaker_notebook_instance

TEST: tests/aws_sagemaker_notebook_instance
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_partition.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_iam_role.my_role: Creating...
aws_iam_role.my_role: Creation complete after 5s [id=turbottest44486]
aws_sagemaker_notebook_instance.named_test_resource: Creating...
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [10s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [20s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [30s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [40s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [50s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [1m0s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [1m10s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [1m20s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [1m30s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [1m40s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [1m50s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [2m0s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [2m10s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [2m20s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [2m30s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [2m40s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [2m50s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [3m0s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [3m10s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Still creating... [3m20s elapsed]
aws_sagemaker_notebook_instance.named_test_resource: Creation complete after 3m26s [id=turbottest44486]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

account_id = 986325076436
aws_partition = aws
aws_region = us-east-1
resource_aka = arn:aws:sagemaker:us-east-1:986325076436:notebook-instance/turbottest44486
resource_id = turbottest44486
resource_name = turbottest44486

Running SQL query: test-get-query.sql
[
  {
    "account_id": "986325076436",
    "arn": "arn:aws:sagemaker:us-east-1:986325076436:notebook-instance/turbottest44486",
    "name": "turbottest44486",
    "partition": "aws",
    "region": "us-east-1",
    "tags": {
      "name": "turbottest44486"
    }
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "akas": [
      "arn:aws:sagemaker:us-east-1:986325076436:notebook-instance/turbottest44486"
    ],
    "name": "turbottest44486",
    "tags": {
      "name": "turbottest44486"
    }
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:sagemaker:us-east-1:986325076436:notebook-instance/turbottest44486",
    "name": "turbottest44486",
    "partition": "aws",
    "region": "us-east-1"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:sagemaker:us-east-1:986325076436:notebook-instance/turbottest44486"
    ],
    "region": "us-east-1",
    "tags": {
      "name": "turbottest44486"
    },
    "title": "turbottest44486"
  }
]
✔ PASSED

POSTTEST: tests/aws_sagemaker_notebook_instance

TEARDOWN: tests/aws_sagemaker_notebook_instance

SUMMARY:

1/1 passed.

```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
N/A
```
</details>

### CODE REVIEW

File: aws/table_aws_sagemaker_notebook_instance.go [hunk 1]
```diff
@@ -249,7 +249,13 @@ func listAwsSageMakerNotebookInstanceTags(ctx context.Context, d *plugin.QueryDa
...
 	}
 
-	resourceArn := h.Item.(*sagemaker.DescribeNotebookInstanceOutput).NotebookInstanceArn
+	var resourceArn string
+	switch h.Item.(type) {
+	case *sagemaker.NotebookInstanceSummary:
+		resourceArn = *h.Item.(*sagemaker.NotebookInstanceSummary).NotebookInstanceArn
+	case *sagemaker.DescribeNotebookInstanceOutput:
+		resourceArn = *h.Item.(*sagemaker.DescribeNotebookInstanceOutput).NotebookInstanceArn
+	}
```
Comment (Subhajit97) [inline aws/table_aws_sagemaker_notebook_instance.go:258]: Can we add a separate function for this?
also, do we need the following check for GET call?
https://github.com/turbot/steampipe-plugin-aws/pull/372/files#diff-2ceeebe3d446a630c976a1a76ab24829e58568b0645e3ae8e4aaa5390a19e234R214-R219

File: aws/table_aws_sagemaker_notebook_instance.go [hunk 2]
```diff
@@ -292,12 +292,12 @@ func getAwsSageMakerNotebookInstanceTurbotTags(ctx context.Context, d *transform
 	return nil, nil
 }
 
-func instanceName(item interface{}) string {
+func notebookInstanceArn(item interface{}) string {
```
Comment (Subhajit97) [inline aws/table_aws_sagemaker_notebook_instance.go:295]:
```suggestion
func notebookInstanceARN(item interface{}) string {
```

File: aws/table_aws_sagemaker_notebook_instance.go [hunk 3]
```diff
@@ -276,7 +276,7 @@ func listAwsSageMakerNotebookInstanceTags(ctx context.Context, d *plugin.QueryDa
 
 func getAwsSageMakerNotebookInstanceTurbotTags(ctx context.Context, d *transform.TransformData) (interface{},
 	error) {
-		data := d.HydrateItem.(*sagemaker.ListTagsOutput)
+	data := d.HydrateItem.(*sagemaker.ListTagsOutput)
 
 	if data.Tags == nil {
```
Comment (Subhajit97) [inline aws/table_aws_sagemaker_notebook_instance.go:281]: Do we need this block, as we returning nil at the end?
https://github.com/turbot/steampipe-plugin-aws/blob/f2b280097bc5e11414422ec75eea864f740c9eaa/aws/table_aws_sagemaker_notebook_instance.go#L281-L283

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@Priyanka585464 There are few changes required, otherwise it looks good. Please see comments, thanks!

**Review by Subhajit97 — APPROVED** [review]
LGTM

File: aws-test/tests/aws_sagemaker_notebook_instance/test-notfound-query.sql [hunk 4]
```diff
@@ -1,3 +1,3 @@
 select title, akas, tags, region, account_id
 from aws.aws_sagemaker_notebook_instance
-where name = '{{ resourceName }}';
\ No newline at end of file
+where name = '{{ resourceName }}3p';
```
Comment (cbruno10) [inline aws-test/tests/aws_sagemaker_notebook_instance/test-notfound-query.sql:3]: @Priyanka585464 Why does this resource name have 3p appended on?

**Review by cbruno10 — CHANGES_REQUESTED** [review]
Please see comment

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**